tasks/coverage/parser_typescript.snap

```diff
@@ -4639,6 +4639,19 @@ Expect to Parse: "compiler/withStatementInternalComments.ts"
  2 │ /*1*/ with /*2*/ ( /*3*/ false /*4*/ ) /*5*/ {}
    ·       ────
    ╰────
+Expect to Parse: "conformance/async/es6/asyncWithVarShadowing_es6.ts"
+
+  × Identifier `x` has already been declared
+     ╭─[conformance/async/es6/asyncWithVarShadowing_es6.ts:130:14]
+ 129 │     }
+ 130 │     catch ({ x }) {
+     ·              ┬
+     ·              ╰── `x` has already been declared here
+ 131 │         var x;
+     ·             ┬
+     ·             ╰── It can not be redeclared here
+ 132 │     }
+     ╰────
```

Comment:
It is correct

https://esbuild.github.io/try/#dAAwLjIzLjAALS1taW5pZnkAYXN5bmMgZnVuY3Rpb24gZm4yNih4KSB7CiAgICB0cnkgewoKICAgIH0KICAgIGNhdGNoICh7IHggfSkgewogICAgICAgIHZhciB4OwogICAgfQp9

Reply:
I remember this one! I was so confused :-/

Reply:
What?? In NodeJS, this runs without error:

```js
try {
    throw { x: 123 };
} catch (x) {
    var x = 456;
}
```

and this throws `SyntaxError: Identifier 'x' has already been declared`:

```js
try {
    throw { x: 123 };
} catch ({ x }) {
    var x = 456;
}
```

But look what this does:

```js
try {
    throw {x: 123};
} catch (x) {
    console.log(x); // Logs {x: 123}
    var x = 456;
    console.log(x); // Logs 456
    var y = 789;
}
console.log(x); // Logs undefined (!!!)
console.log(y); // Logs 789
```

Note that the last `console.log(x)` does NOT throw `ReferenceError: x is not defined`. So `var x` has created a hoisted binding in outer scope, but it's not initialized (unlike `var y = 789;` which creates binding *and* sets its value in outer scope). Is this correct behavior??

Reply:
I can't explain why this is the case, from the example it seems to be because `CatchClause` can't change the value of the `CatchParameter` internally

Reply:
Read the spec. It tried to fix for es2015.